nautobot/extras/models/customfields.py

```diff
@@ -724,7 +724,17 @@ def delete(self, *args, **kwargs):
 
         super().delete(*args, **kwargs)
 
-        delete_custom_field_data.delay(self.key, content_types)
+        # Circular Import
+        from nautobot.extras.signals import change_context_state
+
+        change_context = change_context_state.get()
+        if change_context is None:
+            context = None
+        else:
+            context = change_context.as_dict()
+            context["user"] = change_context.get_user(self)
```

Comment:
Can we consolidate this even a bit further? A la:

```suggestion
            context = change_context.as_dict(instance=self)
```